library/Neovim/Plugin/Classes.hs

```diff
@@ -447,3 +448,7 @@ instance HasFunctionName FunctionalityDescription where
         Command   n _ -> n
         Autocmd _ n _ _ -> n
 
+    methodName = \case
+        Function (F n) _ -> "function:" <> n
+        Command (F n) _ -> "command:" <> n
```

Comment:
This may be premature optimization, but how about suffixing it? Most lookups in the map will needlessly compare the prefix for every comparison, so adding it as a suffix seems to make more sense to me. E.g. `(<> " :: function")` Since this addition is kind of like a type, I kinda like it that way. :-)

Reply:
The whitespaces may be a problem, though.

Reply:
sure, I just replicated what I did for my python plugin provider. not sure about the whitespaces, I think neovim should also just use them as strings

Reply:
and the single colon is how neovim scopes the identifiers